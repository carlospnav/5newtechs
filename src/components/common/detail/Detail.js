import styled from "styled-components"
import css from "theme/variables"

export const Value = styled.h2`
  font-family: ${css.font.roboto};
  font-size: 22px;
  font-weight: ${css.fontWeight.regular};
  text-align: left;
  letter-spacing: 0.8px;
  color: white;
  margin: 0;
`
export const Label = styled.h4`
  font-family: ${css.font.roboto};
  font-size: 14px;
  font-weight: ${css.fontWeight.regular};
  color: ${css.color.sectionTitle};
  margin: 0;
  letter-spacing: 0.5px;
`

export const SpacedDetail = styled.div`
  margin-right: 20px;
`
export const RegularDetail = styled.div`
  width: 130px;
  margin-bottom: 10px;
`

export default function({spaced, value, label}){ 
  const valueContent = (
    <Value>{spaced ? value : value.toUpperCase()}</Value> 
  )
  const labelContent = ( 
    <Label>{label}</Label>
  )

  return (
    spaced ? (
      <SpacedDetail>
        {valueContent}
        {labelContent}
      </SpacedDetail>
    ) : (
      <RegularDetail>
        {valueContent}
        {labelContent}
      </RegularDetail>
    )
  )
}