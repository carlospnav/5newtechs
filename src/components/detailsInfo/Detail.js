import styled from "styled-components"
import css from "theme/variables"

const Detail = styled.div`
  width: 130px;
  margin-bottom: 10px;
`
export default function({children}){ 
  return (
    <Detail>{children}</Detail>
  )
}