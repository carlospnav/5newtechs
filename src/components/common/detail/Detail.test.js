import React from "react"
import { shallow } from "enzyme"
import Detail, { Value, Label, SpacedDetail, RegularDetail } from "./Detail"

describe("<Detail>", () => {
  let wrapper;

  function detail(){
    wrapper = shallow(<Detail spaced={true} value="Valor" label="Label"/>)
  }

  beforeEach(detail)

  afterEach(() => wrapper = undefined)

  it("renders a spaced detail component if 'spaced' is passed as props", () => {
    expect(wrapper.find(SpacedDetail)).toHaveLength(1)
  })

  it("renders a Value component with value props as UPPERCASED text.", () => {
    expect(wrapper.find(Value).dive().text()).toEqual("VALOR")
  })

  it("renders a Label component with label props as text.", () => {
    expect(wrapper.find(Label).dive().text()).toEqual("Label")
  })
})