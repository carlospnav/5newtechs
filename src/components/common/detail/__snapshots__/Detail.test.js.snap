// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail> renders a regular detail component if 'spaced' is NOT passed as props 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  font-weight: 300;
  text-align: left;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
  color: white;
  margin: 0;
}

.c2 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  font-weight: 300;
  color: #4e5760;
  margin: 0;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c0 {
  width: 130px;
  margin-bottom: 10px;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.h2>
      <h2
        className="c1"
      >
        VALOR
      </h2>
    </styled.h2>
    <styled.h4>
      <h4
        className="c2"
      >
        Label
      </h4>
    </styled.h4>
  </div>
</styled.div>
`;

exports[`<Detail> renders a spaced detail component if 'spaced' is passed as props 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  font-weight: 300;
  text-align: left;
  -webkit-letter-spacing: 0.8px;
  -moz-letter-spacing: 0.8px;
  -ms-letter-spacing: 0.8px;
  letter-spacing: 0.8px;
  color: white;
  margin: 0;
}

.c2 {
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  font-weight: 300;
  color: #4e5760;
  margin: 0;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c0 {
  margin-right: 20px;
}

<styled.div>
  <div
    className="c0"
  >
    <styled.h2>
      <h2
        className="c1"
      >
        Valor
      </h2>
    </styled.h2>
    <styled.h4>
      <h4
        className="c2"
      >
        Label
      </h4>
    </styled.h4>
  </div>
</styled.div>
`;
